File: 3761-minimum-absolute-distance-between-mirror-pairs/3761-minimum-absolute-distance-between-mirror-pairs.py
class Solution:
    def minMirrorPairDistance(self, nums: List[int]) -> int:
        def rev(x):
            r = 0
            while x > 0:
                r = r * 10 + x % 10
                x //= 10
            return r
        last = {}
        mind = float('inf')
        for j in range(len(nums)):
            if nums[j] in last:
                mind = min(mind, j - last[nums[j]])
            rj = rev(nums[j])
            last[rj] = j
        return -1 if mind == float('inf') else mind

File: 3761-minimum-absolute-distance-between-mirror-pairs/3761-minimum-absolute-distance-between-mirror-pairs.cpp
class Solution {
public:
    int minMirrorPairDistance(vector<int>& nums) {
        int n = nums.size();
        int minDist = INT_MAX;
        unordered_map<int, int> mp;
        
        for (int j = 0; j < n; ++j) {
            if (mp.find(nums[j]) != mp.end()) {
                minDist = min(minDist, j - mp[nums[j]]);
            }
            
            long long rev = 0;
            int temp = nums[j];
            while (temp > 0) {
                rev = rev * 10 + temp % 10;
                temp /= 10;
            }
            
            mp[(int)rev] = j;
        }
        
        return minDist == INT_MAX ? -1 : minDist;
    }
};